### ISSUE TITLE
Configuration options should allow to be configured via connection string

### ISSUE THREAD

**Issue by brettallred** [mxenabled/action_subscriber#13]
Currently the only supported method of configuring Bunny or MarchHare is by a configuration parameters and not by a connection string.

**Comment by mmmries** [comment]
Are you thinking we parse the connection string with the `URI` module and then set the corresponding settings in the configuration parameters? That seems like a pretty simple addition.

**Comment by brettallred** [comment]
```ruby
      def connection_string
        AMQ::Settings.parse_amqp_url(::ActionSubscriber.configuration.connection_string)
      end
```

**Comment by liveh2o** [comment]
I'm for it!

![ezgif-3518449088](https://cloud.githubusercontent.com/assets/207933/9806811/0cf4b8dc-5807-11e5-9634-735f9064f5b4.gif)

**Comment by mmmries** [comment]
We don't have `AMQ` available to us, in this project. That is the eventmachine based implementation isn't it?

**Comment by brettallred** [comment]
Bunny has a dependency on amq-protocol-2.0.0.   This is found in that gem.

bunny (2.2.0)
    amq-protocol (>= 2.0.0)

**Comment by mmmries** [comment]
Nice. Sorry for the confusion.

Instead of checking for the `connection_string` when we are about to use the configuration, what if we just exploded the connection string and set the corresponding settings during the configuration step?

```ruby
module ActionSubscriber
  class Configuration
    ...
    def connection_string=(url)
      settings = ::AMQ::Settings.parse_amqp_url(url)
      settings.each do |key, value|
        send("#{key}=", value)
      end
    end
    ...
  end
end
```

**Comment by brettallred** [comment]
:+1: I like it.

**Comment by liveh2o** [comment]
Closed by #45.
